app/controllers/spree/adyen_redirect_controller.rb

```diff
@@ -27,11 +27,25 @@ def authorise3d
       payment_method = payment.payment_method
       payment.request_env = request.env
       payment_method.perform_authorization_3d(payment, adyen_3d_params)
-      order.next
+      advance_order(order, "confirm")
       redirect_to checkout_state_path(order.state)
     end
 
     private
+
+    def state_before_current?(order, state)
+      steps = order.checkout_steps
+      steps.index(state) < (steps.index(order.state) || 0)
+    end
+
+    def advance_order(order, end_state)
+      return if state_before_current?(order, end_state)
```

Comment:
What happens if end_state is not a valid state? does this error or just keep going to completion?